src/player.cpp

```diff
@@ -4943,38 +4943,22 @@ void player::healall(int dam)
     }
 }
 
-void player::hurtall(int dam)
+void player::hurtall(int dam, Creature *source)
 {
     for (int i = 0; i < num_hp_parts; i++) {
-        hp_cur[i] -= dam;
-        if (hp_cur[i] < 0) {
-            lifetime_stats()->damage_taken += hp_cur[i];
-            hp_cur[i] = 0;
-        }
-        mod_pain( dam / 2 );
-        lifetime_stats()->damage_taken += dam;
+        const body_part bp = hp_to_bp( static_cast<hp_part>( i ) );
+        apply_damage( source, bp, dam );
     }
 }
 
-void player::hitall(int dam, int vary)
+void player::hitall(int dam, int vary, Creature *source)
 {
-    if (in_sleep_state()) {
-        wake_up();
-    }
-
     for (int i = 0; i < num_hp_parts; i++) {
-        int ddam = vary? dam * rng (100 - vary, 100) / 100 : dam;
+        const body_part bp = hp_to_bp( static_cast<hp_part>( i ) );
+        int ddam = vary ? dam * rng (100 - vary, 100) / 100 : dam;
         int cut = 0;
-        absorb((body_part) i, ddam, cut);
-        hp_cur[i] -= ddam;
-        if (hp_cur[i] < 0) {
-            lifetime_stats()->damage_taken += hp_cur[i];
-            hp_cur[i] = 0;
-        }
-
-        // Average of pre and post armor damage levels, divided by 8.
-        mod_pain( (dam + ddam) / 16 );
```

Comment:
Yes, the whole removed code is handled in `apply_damage` (and more), I figured, why the repetition. And there are slight (and pointless) differences: `hitall` wakes you up, but `hurtall` does not.